dev: add basic document describing coverage analysis and how to generate it (#7637)
ArduPilot/ardupilot_wiki: dev/source/docs/coverage-analysis.rst

.. _coverage-analysis:

=================
Coverage Analysis
=================

> ArduPilot has facility to easily run automated checks to generate coverage anal;ysis.

Reviewer: ```suggestion
ArduPilot provides an easy way to run automated checks to generate coverage analysis.
```

Author: Fixed, thanks!

These are used to produce a `periodically-updated report <https://firmware.ardupilot.org/coverage/>`__ on our firmware server.

coveralls.io also uses the same mechanisms to generate its `own report <https://coveralls.io/github/ArduPilot/ardupilot>`__

-------------------------
Running Coverage Analysis
-------------------------

Run the full suite with:

::

   ./Tools/scripts/run_coverage.py --full

This generates data files describing the coverage, but also creates a HTML report for easy perusal.

::

   firefox reports/lcov-report/index.html

----------------------
Improving the coverage
----------------------

> Coverage is generated from the :ref:`autotest suite <the-ardupilot-autotest-framework>`, :ref:`unit tests <unit-tests>` and the :ref:`example sketches <learning-ardupilot-the-example-sketches>`.

Reviewer 2: unit-tests    is not a valid label

try ardupilot-unit-tests

Author: Fixed
